src/main/java/com/example/tournament/Service/FileSaveService.java

```diff
@@ -0,0 +1,23 @@
+package com.example.tournament.Service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.*;
+
+@Service
+public class FileSaveService {
+
+    public void saveToFile(String fileName, String stringObject) throws IOException {
```

Comment:
This component should know how to serialize entity into internal representation, so instead of String please declare here Team/Match/Tournament